### PR TITLE
[tune] Search Alg delay import + CLI timing test

### PR DESCRIPTION
**Pull request by adizim**
Consistent with incoming PR #3983

### CODE REVIEW

**Comment by richardliaw** [comment]
Hm, I think maybe we should think of a better solution for this. 

The objective here is to make sure `tune list-trials` is fast, and doesn't require 8 seconds of imports.

To achieve this, I did a delayed import. However, another option is to remove all of these things from `tune/suggest/__init__.py`. This should make it so that all of these third-party packages will not be imported automatically.

Do you get what I mean?

File: python/ray/tune/suggest/nevergrad.py [hunk 1]
```diff
@@ -3,9 +3,9 @@
 from __future__ import print_function
 
 try:
-    import nevergrad
+    import nevergrad as ng
```
Comment (richardliaw) [inline python/ray/tune/suggest/nevergrad.py]: can you also revert the `_import_` changes I made in #3983?

**Comment by adizim** [comment]
@richardliaw are these changes breaking the examples for you (ImportError)?

**Comment by richardliaw** [comment]
Yeah, 
```
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
Traceback (most recent call last):
  File "/ray/python/ray/tune/examples/bayesopt_example.py", line 12, in <module>
    from ray.tune.suggest import BayesOptSearch
ImportError: cannot import name BayesOptSearch
```

I think you'll need to change the examples to:
```
from ray.tune.suggest.bayesopt import BayesOptSearch
```

File: python/ray/tune/suggest/bayesopt.py [hunk 2]
```diff
@@ -6,13 +6,10 @@
...
 
-byo = None
-
-
-def _import_bayesopt():
-    global byo
-    import bayes_opt
-    byo = bayes_opt
+try:
+    import bayesopt as byo
```
Comment (richardliaw) [inline python/ray/tune/suggest/bayesopt.py]: I think you instead now don't need to wrap these if we commit this change

```
import bayesopt as byo
```

File: python/ray/tune/suggest/__init__.py [hunk 3]
```diff
@@ -1,22 +1,12 @@
 from ray.tune.suggest.search import SearchAlgorithm
 from ray.tune.suggest.basic_variant import BasicVariantGenerator
 from ray.tune.suggest.suggestion import SuggestionAlgorithm
-from ray.tune.suggest.bayesopt import BayesOptSearch
-from ray.tune.suggest.hyperopt import HyperOptSearch
-from ray.tune.suggest.nevergrad import NevergradSearch
-from ray.tune.suggest.skopt import SkOptSearch
-from ray.tune.suggest.sigopt import SigOptSearch
```
Comment (richardliaw) [inline python/ray/tune/suggest/__init__.py]: for all of these, can we also in this file put something like

```python
def SigOptSearch(*args, **kwargs):
    raise DeprecationWarning("This class has been moved. Please import via `from ray.tune.suggest.skopt import SkOptSearch`")
```

File: python/ray/tune/suggest/hyperopt.py [hunk 4]
```diff
@@ -9,15 +9,13 @@
 from ray.tune.error import TuneError
 from ray.tune.suggest.suggestion import SuggestionAlgorithm
 
-hpo = None
+try:
+    import hyperopt as hpo
```
Comment (richardliaw) [inline python/ray/tune/suggest/hyperopt.py]: same here - (and in below files) `import hyperopt as hpo`.

**Comment by richardliaw** [comment]
(don't forget to run the linting script in ray/scripts/format.sh)

**Comment by adizim** [comment]
@richardliaw all the example work except BayesOpt, but I think it was broken for a long time. Tried to diagnose the issue and it seems that the byo.BayesianOptimization object has no suggest method.

**Review by richardliaw — APPROVED** [review]

**Comment by richardliaw** [comment]
@adizim I had to revert some changes because it was failing a lint test.

**Comment by richardliaw** [comment]
Thanks for doing this, @adizim!

**Pull request merged**